### PR TITLE
Add test for gateway.go 

### LINKED ISSUE #488: Add tests to clients package
Currently, we [only test](https://github.com/NethermindEth/juno/blob/develop/clients/gateway_test.go) that json responses properly marshal into our corresponding struct types. We lack coverage for the actual business logic.

The task is to add tests for the client package (specifically `gateway.go`) by following the testing process mentioned in [Learn Go With Tests](https://quii.gitbook.io/learn-go-with-tests/). There is a HTTP server section which describes how to write tests using httptest package. For coverage stats and visuals, use `make test-cover`.

### PR DESCRIPTION
**Pull request by liusanchuan**
Fixes  #488 
## Description

Add test to cover `gateway.go`

## Changes:

- Add test in `juno/clients/gateway_test.go`

## Types of changes

- Other (please describe):  Add more test

## Testing

**Requires testing**: Yes

**Did you write tests??**: Yes

## Documentation

**If this requires a documentation update, did you add one?** No

### CODE REVIEW

**Comment by liusanchuan** [comment]
@joshklop 
please make a review
ok      github.com/NethermindEth/juno/clients   0.085s  coverage: 100.0% of statements

**Review by joshklop — CHANGES_REQUESTED** [review]
Thanks! To start, can you please break up the single monolithic test into multiple test cases? See Learn Go with Tests (liked in corresponding issue) for examples.

File: clients/gateway_test.go [hunk 1]
```diff
@@ -287,3 +290,286 @@ func TestClassUnmarshal(t *testing.T) {
...
+	t.Run("Test normal case", func(t *testing.T) {
+		stateUpdate, err := gatewayClient.GetStateUpdate(10)
+		assert.Equal(t, nil, err, "Unexpected error")
+		assert.Equal(t, update, *stateUpdate)
+	})
+	t.Run("Test block number out of boundary", func(t *testing.T) {
+		stateUpdate, err := gatewayClient.GetStateUpdate(1000000)
+		assert.Nil(t, stateUpdate, "Unexpected error")
+		assert.NotNil(t, err)
+	})
```
Comment (liusanchuan) [inline clients/gateway_test.go:381]: @joshklop  Do you think should this format split into multiple test function?

**Comment by liusanchuan** [comment]
@joshklop ,  Thanks for your point out. Please review my fix code.

**Review by IronGauntlets — APPROVED** [review]
LGTM!

Please fix the conflicts and clean the git history and follow the commit message guidelines mentioned [here](https://cbea.ms/git-commit/). Or you can squash and merge but please make sure to update the git commit message according to the guide mentioned above.

**Review by joshklop — APPROVED** [review]
Fantastic stuff. Thanks @liusanchuan!

**Comment by liusanchuan** [comment]
Hi @IronGauntlets , I have squeezed my branch comments :) , could you have a review?

**Comment by liusanchuan** [comment]
@joshklop @IronGauntlets Thanks for your guide, I learned a lot from this cooperation. If there is others tasks, welcome to assign to me. :)

**Pull request merged**